coalib/bearlib/aspects/collections.py

```diff
@@ -16,6 +16,13 @@ def __init__(self, seq=()):
         list.__init__(self, seq)
 
     def __contains__(self, aspect):
+        if hasattr(self, 'bear') and isinstance(aspect, aspectbase):
+            compatible = False
+            for language in self.bear.LANGUAGES:
+                if str(aspect.language).split()[0] == language:
+                    compatible = True
+            if not compatible:
+                return False
         for item in self:
             if issubaspect(aspect, item):
```

Comment:
One further remark: You should move the whole language match logic to here, so you know first if the `aspect` is applicable at all for the bear. And you implicitly know that the incoming `aspect` object is either an aspect class or instance and can simplify `if isinstance(aspect, aspectbase)` to just `if not isclass(aspect)`